src/cmap/establish/handshake/mod.rs

```diff
@@ -150,6 +157,10 @@ impl Handshaker {
         let mut command =
             is_master_command(options.as_ref().and_then(|opts| opts.server_api.as_ref()));
 
+        let handler = options
+            .as_ref()
+            .and_then(|options| options.sdam_event_handler.clone());
```

Comment:
Very minor style suggestion: I tend to explicitly write `Arc::clone(...)` to make it clear at the site that it's the pointer being cloned, not the value.